### PR TITLE
bi-directional transformer rescoring

### PR DESCRIPTION
**Pull request by qingerVT**
Summary:
rescore the generated sentences by bi-directioanl transformer. The idea is predict every word given all surronding words except itself, i.e., p(y_t | y_{\neq t}).
The training strategy is quite like BERT, but don't need to mask any words to avoid self-attention when we only have a single layer (check out FAIR's paper: "Cloze-driven Pretraining of Self-attention Networks")

The loss function keeps unchanged, such as cross entropy. The score of a given hypothesis is \sum log p(y_t | y_{\neq t})

Differential Revision: D14926425

### CODE REVIEW

**Comment by qingerVT** [comment]
Haha, Awesome question, Stefan!! Hope this paper can be published very soon!! :-)

**Comment by michaelauli** [comment]
@stefan-it You can find the code here: https://github.com/pytorch/fairseq/tree/
bi_trans_lm/examples/pretraining

It is part of the release for another paper.
